components/server/src/auth/auth-provider-service.ts

```diff
@@ -16,17 +16,21 @@ import { oauthUrls as bbsUrls } from "../bitbucket-server/bitbucket-server-urls"
 import { oauthUrls as bbUrls } from "../bitbucket/bitbucket-urls";
 import { log } from "@gitpod/gitpod-protocol/lib/util/logging";
 import fetch from "node-fetch";
+import { Authorizer } from "../authorization/authorizer";
+import { ApplicationError, ErrorCodes } from "@gitpod/gitpod-protocol/lib/messaging/error";
+import { GitHubScope } from "../github/scopes";
+import { GitLabScope } from "../gitlab/scopes";
+import { BitbucketOAuthScopes } from "../bitbucket/bitbucket-oauth-scopes";
+import { BitbucketServerOAuthScopes } from "../bitbucket-server/bitbucket-server-oauth-scopes";
 
 @injectable()
 export class AuthProviderService {
```

Comment:
Maybe we start adding DB tests on this level?

Reply:
I'm on it right now.